strings/base_coroutine_foundation.h

```diff
@@ -454,9 +461,20 @@ namespace winrt::impl
         {
             slim_lock_guard const guard(m_lock);
 
-            if (m_status.load(std::memory_order_relaxed) == AsyncStatus::Completed)
+            if constexpr (std::is_same_v<TProgress, void>)
+            {
+                if (m_status.load(std::memory_order_relaxed) == AsyncStatus::Completed)
+                {
+                    return static_cast<Derived*>(this)->get_return_value();
+                }
+            }
+            else
             {
-                return static_cast<Derived*>(this)->get_return_value();
+                if (m_status.load(std::memory_order_relaxed) == AsyncStatus::Completed ||
```

Comment:
Is it OK to load twice here?

Reply:
Ah, I didn't spot that. It's safe but I think it would be cleaner and cheaper to load once and compare the result.

Reply:
Ooh, good catch. Should load once and test the cached copy. (std::atomic.load() forces a reach back to memory each you load.)

Reply:
Turns out there are a lot of places where I perform redundant loads from the atomic.